File: TICTACTOE.cpp
#include <iostream>
#include <windows.h>
using namespace std;

int main()
{
    start:
    char arr[3][3] = {{' ',' ',' '},{' ',' ',' '},{' ',' ',' '}};
    int counter = 1;
    while(!0)
    {
        system("cls");
        cout << "     |     |     \n";
        cout << "  " << arr[0][0] <<"  |" << "  " << arr[0][1] <<"  |" << "  " << arr[0][2] <<"  \n";
        cout << "_____|_____|_____\n";
        cout << "     |     |     \n";
        cout << "  " << arr[1][0] <<"  |" << "  " << arr[1][1] <<"  |" << "  " << arr[1][2] <<"  \n";
        cout << "_____|_____|_____\n";
        cout << "     |     |     \n";
        cout << "  " << arr[2][0] <<"  |" << "  " << arr[2][1] <<"  |" << "  " << arr[2][2] <<"  \n";
        cout << "     |     |     \n";
        
        if (counter%2==1)
        {  
            counter ++;
            int x, y;
            cout << "PLAYER X, ENTER YOUR COORDINATES\n";
            cout << "row(0-2):\n";
            cin >> x;
            cout << "column(0-2):\n";
            cin >> y;
            if (x>=0 && x<=2 && y>=0 && y<=2 && arr[x][y]==' ')
                arr[x][y] = 'X';
            else
                {
                    cout << "WARNING! USE UNPOPULATED SLOTS ONLY AND KEEP THE COORDINATES IN RANGE\n";
                    cout << "PLAYER X, ENTER YOUR COORDINATES\n";
                    cout << "row(0-2):\n";
                    cin >> x;
                    cout << "column(0-2):\n";
                    cin >> y;
                    if (x>=0 && x<=2 && y>=0 && y<=2 && arr[x][y]==' ')
                        arr[x][y] = 'X';
                };


            system("cls");
            cout << "     |     |     \n";
            cout << "  " << arr[0][0] <<"  |" << "  " << arr[0][1] <<"  |" << "  " << arr[0][2] <<"  \n";
            cout << "_____|_____|_____\n";
            cout << "     |     |     \n";
            cout << "  " << arr[1][0] <<"  |" << "  " << arr[1][1] <<"  |" << "  " << arr[1][2] <<"  \n";
            cout << "_____|_____|_____\n";
            cout << "     |     |     \n";
            cout << "  " << arr[2][0] <<"  |" << "  " << arr[2][1] <<"  |" << "  " << arr[2][2] <<"  \n";
            cout << "     |     |     \n";

            if ((arr[0][0]==arr[0][1] && arr[0][1]=='X' && arr[0][1]==arr[0][2]) ||
                (arr[1][0]==arr[1][1] && arr[1][1]=='X' && arr[1][1]==arr[1][2]) ||
                (arr[2][0]==arr[2][1] && arr[2][1]=='X' && arr[2][1]==arr[2][2]) ||
                (arr[0][0]==arr[1][0] && arr[1][0]=='X' && arr[1][0]==arr[2][0]) ||
                (arr[0][1]==arr[1][1] && arr[1][1]=='X' && arr[1][1]==arr[2][1]) ||
                (arr[0][2]==arr[1][2] && arr[1][2]=='X' && arr[1][2]==arr[2][2]) ||
                (arr[0][0]==arr[1][1] && arr[1][1]=='X' && arr[1][1]==arr[2][2]) ||
                (arr[0][2]==arr[1][1] && arr[1][1]=='X' && arr[1][1]==arr[2][0]) )
                {
                    cout << "PLAYER X WINS\n";
                    Sleep(3000);
                    system("cls");
                    goto start;

                }
            else if (arr[0][0]!=' ' &&
                     arr[0][1]!=' ' &&
                     arr[0][2]!=' ' &&
                     arr[1][0]!=' ' &&
                     arr[1][1]!=' ' &&
                     arr[1][2]!=' ' &&
                     arr[2][0]!=' ' &&
                     arr[2][1]!=' ' &&
                     arr[2][2]!=' ')
                     {
                        cout << "NOBODY WINS\n";
                        Sleep(3000);
                        goto start;

                     };
        };
        if (counter%2==0)
        {
            counter ++;
            int x, y;
            cout << "PLAYER O, ENTER YOUR COORDINATES\n";
            cout << "row(0-2):\n";
            cin >> x;
            cout << "column(0-2):\n";
            cin >> y;
            if (x>=0 && x<=2 && y>=0 && y<=2 && arr[x][y]==' ')
                arr[x][y] = 'O';
            else
                {
                    cout << "WARNING! USE UNPOPULATED SLOTS ONLY AND KEEP THE COORDINATES IN RANGE\n";
                    cout << "PLAYER O, ENTER YOUR COORDINATES\n";
                    cout << "row(0-2):\n";
                    cin >> x;
                    cout << "column(0-2):\n";
                    cin >> y;
                    if (x>=0 && x<=2 && y>=0 && y<=2 && arr[x][y]==' ')
                        arr[x][y] = 'O';
                };

            system("cls");
            cout << "     |     |     \n";
            cout << "  " << arr[0][0] <<"  |" << "  " << arr[0][1] <<"  |" << "  " << arr[0][2] <<"  \n";
            cout << "_____|_____|_____\n";
            cout << "     |     |     \n";
            cout << "  " << arr[1][0] <<"  |" << "  " << arr[1][1] <<"  |" << "  " << arr[1][2] <<"  \n";
            cout << "_____|_____|_____\n";
            cout << "     |     |     \n";
            cout << "  " << arr[2][0] <<"  |" << "  " << arr[2][1] <<"  |" << "  " << arr[2][2] <<"  \n";
            cout << "     |     |     \n";

            if ((arr[0][0]==arr[0][1] && arr[0][1]=='O' && arr[0][1]==arr[0][2]) ||
                (arr[1][0]==arr[1][1] && arr[1][1]=='O' && arr[1][1]==arr[1][2]) ||
                (arr[2][0]==arr[2][1] && arr[2][1]=='O' && arr[2][1]==arr[2][2]) ||
                (arr[0][0]==arr[1][0] && arr[1][0]=='O' && arr[1][0]==arr[2][0]) ||
                (arr[0][1]==arr[1][1] && arr[1][1]=='O' && arr[1][1]==arr[2][1]) ||
                (arr[0][2]==arr[1][2] && arr[1][2]=='O' && arr[1][2]==arr[2][2]) ||
                (arr[0][0]==arr[1][1] && arr[1][1]=='O' && arr[1][1]==arr[2][2]) ||
                (arr[0][2]==arr[1][1] && arr[1][1]=='O' && arr[1][1]==arr[2][0]) )
                {
                    cout << "PLAYER O WINS\n";
                    Sleep(3000);
                    system("cls");
                    goto start;

                }
            else if (arr[0][0]!=' ' &&
                     arr[0][1]!=' ' &&
                     arr[0][2]!=' ' &&
                     arr[1][0]!=' ' &&
                     arr[1][1]!=' ' &&
                     arr[1][2]!=' ' &&
                     arr[2][0]!=' ' &&
                     arr[2][1]!=' ' &&
                     arr[2][2]!=' ')
                     {
                        cout << "NOBODY WINS\n";
                        Sleep(3000);
                        goto start;

                     };
        };




    };





    

    return 0;
};

File: Tic Tac Toe.py
import os
from time import *


arr = [[' ',' ',' '],[' ',' ',' '],[' ',' ',' ']]
counter = 1
while not False:
    os.system('cls')
    print("     |     |     ")
    print(" ", arr[0][0], " | ", arr[0][1], " | ", arr[0][2])
    print("_____|_____|_____")
    print("     |     |     ")
    print(" ", arr[1][0], " | ", arr[1][1], " | ", arr[1][2])
    print("_____|_____|_____")
    print("     |     |     ")
    print(" ", arr[2][0], " | ", arr[2][1], " | ", arr[2][2])
    print("     |     |     ")


    if counter%2==1:
        counter+=1
        x = int(input("PLAYER X, ENTER YOUR COORDINATES\nrow(0-2): "))
        y = int(input("column(0-2): "))
        if arr[x][y]==' ':
            arr[x][y] = 'X'
        else:
            print("WARNING! USE UNPOPULATED SLOTS ONLY")
            x = int(input("PLAYER X, ENTER YOUR COORDINATES\nrow(0-2): "))
            y = int(input("column(0-2): "))
            if arr[x][y]==' ':
                arr[x][y] = 'X'
    
        os.system("cls")
        print("     |     |     ")
        print(" ", arr[0][0], " | ", arr[0][1], " | ", arr[0][2])
        print("_____|_____|_____")
        print("     |     |     ")
        print(" ", arr[1][0], " | ", arr[1][1], " | ", arr[1][2])
        print("_____|_____|_____")
        print("     |     |     ")
        print(" ", arr[2][0], " | ", arr[2][1], " | ", arr[2][2])
        print("     |     |     ")

        if (arr[0][0]==arr[0][1]==arr[0][2]=='X' or
            arr[1][0]==arr[1][1]==arr[1][2]=='X' or
            arr[2][0]==arr[2][1]==arr[2][2]=='X' or
            arr[0][0]==arr[1][0]==arr[2][0]=='X' or
            arr[0][1]==arr[1][1]==arr[2][1]=='X' or
            arr[0][2]==arr[1][2]==arr[2][2]=='X' or
            arr[0][2]==arr[1][1]==arr[2][0]=='X' or
            arr[0][0]==arr[1][1]==arr[2][2]=='X') :
            print("PLAYER X WINS")
            sleep(3)
            os.system("cls")
            arr = [[' ',' ',' '],[' ',' ',' '],[' ',' ',' ']]
            counter = 1
            continue
        
        elif (  arr[0][0]!=' ' and
                arr[0][1]!=' ' and
                arr[0][2]!=' ' and
                arr[1][0]!=' ' and
                arr[1][1]!=' ' and
                arr[1][2]!=' ' and
                arr[2][0]!=' ' and
                arr[2][1]!=' ' and
                arr[2][2]!=' '):
            print("NOBODY WINS")
            sleep(3)
            arr = [[' ',' ',' '],[' ',' ',' '],[' ',' ',' ']]
            counter = 1
            continue

    if counter%2==0:
        counter+=1
        x = int(input("PLAYER O, ENTER YOUR COORDINATES\nrow(0-2): "))
        y = int(input("column(0-2): "))
        if arr[x][y]==' ':
            arr[x][y] = 'O'
        else:
            print("WARNING! USE UNPOPULATED SLOTS ONLY")
            x = int(input("PLAYER O, ENTER YOUR COORDINATES\nrow(0-2): "))
            y = int(input("column(0-2): "))
            if arr[x][y]==' ':
                arr[x][y] = 'O'


        os.system("cls")
        print("     |     |     ")
        print(" ", arr[0][0], " | ", arr[0][1], " | ", arr[0][2])
        print("_____|_____|_____")
        print("     |     |     ")
        print(" ", arr[1][0], " | ", arr[1][1], " | ", arr[1][2])
        print("_____|_____|_____")
        print("     |     |     ")
        print(" ", arr[2][0], " | ", arr[2][1], " | ", arr[2][2])
        print("     |     |     ")

        if (arr[0][0]==arr[0][1]==arr[0][2]=='O' or
            arr[1][0]==arr[1][1]==arr[1][2]=='O' or
            arr[2][0]==arr[2][1]==arr[2][2]=='O' or
            arr[0][0]==arr[1][0]==arr[2][0]=='O' or
            arr[0][1]==arr[1][1]==arr[2][1]=='O' or
            arr[0][2]==arr[1][2]==arr[2][2]=='O' or
            arr[0][2]==arr[1][1]==arr[2][0]=='O' or
            arr[0][0]==arr[1][1]==arr[2][2]=='O') :
            print("PLAYER O WINS")
            sleep(3)
            os.system("cls")
            arr = [[' ',' ',' '],[' ',' ',' '],[' ',' ',' ']]
            counter = 1
            continue

        elif (  arr[0][0]!=' ' and
                arr[0][1]!=' ' and
                arr[0][2]!=' ' and
                arr[1][0]!=' ' and
                arr[1][1]!=' ' and
                arr[1][2]!=' ' and
                arr[2][0]!=' ' and
                arr[2][1]!=' ' and
                arr[2][2]!=' '):
                    print("NOBODY WINS")
                    sleep(3)
                    arr = [[' ',' ',' '],[' ',' ',' '],[' ',' ',' ']]
                    counter = 1
                    continue
  


